# D-BPP workflow

Yang Y, Pang XX, Ding YM, Zhang BW, Bai WN, Zhang DY. 2025. Synergizing Bayesian and Heuristic Approaches: D-BPP Uncovers Ghost Introgression in Panthera and Thuja. bioRxiv 2025.06.27.662067. doi: https://doi.org/10.1101/2025.06.27.662067

To resolve complex evolutionary relationships shaped by introgression, we developed D-BPP—a modular, four-step framework that integrates heuristic and Bayesian methods. 

Step 1:
we assess species-tree uncertainty by identifying as many plausible topologies as we can. Instead of relying on a single fixed species tree—as is often done (e.g., Ji et al. 2023; Ji et al. 2025)—we examine multiple plausible topologies, acknowledging that both ILS and gene flow can obscure phylogenetic signals. In practice, conflicting phylogenies may arise from different genetic markers (e.g., mitochondrial vs. nuclear DNA) and/or analytical methods (e.g., coalescent versus concatenation). With whole-genome data, it is now common to reconstruct sliding-window phylogenies along the genome (e.g., Santos et al. 2025), often yielding multiple plausible topologies that require further evaluation.

Step 2:
For each candidate species-tree topology, we apply the D-statistic (Green et al. 2010; Durand et al. 2011) to all valid species triples in the form ((P1, P2), P3), using an outgroup to polarize derived allele states. Under the null hypothesis of no gene flow, the two discordant site patterns, ‘ABBA’ and ‘BABA’, are expected to occur at equal frequencies. In Dsuite (Malinsky et al. 2021), P1 and P2 are ordered such that the D-statistic is always positive. A non-zero D-statistic, typically resulting from a significant excess of ABBA over BABA, is commonly interpreted as gene flow between P2 and P3—either from P3 to P2 (inflow, Fig. 2A) or vice versa (outflow, Fig. 2B). However, Tricou et al. (Tricou et al. 2022b) demonstrated that introgression into P1 from an unsampled “midgroup” lineage—situated between the ingroup and outgroup—can also produce a significant D-value (Fig. 2C). We further demonstrated that ghost lineages diverging before the outgroup (“basal” ghost lineages, Fig. 2D) or originating from the outgroup itself (“outgroup-derived” ghost lineages, Fig. 2E) can likewise yield significant D-statistic (see Supplementary Text for a mathematical proof). Thus, each significant D-statistic may reflect one of three candidate introgression events, i.e., inflow, outflow, and ghost introgression (involving three distinct relative positions of the ghost lineage with respect to the outgroup; Fig. 2). For each significant triple, we computed D_p=\frac{ABBA-BABA}{BBAA+ABBA+BABA} (Hamlin et al. 2020), a statistic conceptually more aligned with genome-wide introgression proportion (Hibbins and Hahn 2022). Significant triples were ranked in descending order of their Dp values (Fig. 1), facilitating efficient prioritization for downstream testing.

Step 3：
Each significant triple from Step 2 is sequentially evaluated using the Bayesian test (Ji et al. 2023) implemented in BPP (Flouri et al. 2020) to confirm genuine introgression events. The test evaluates support for gene flow (H₁) versus its absence (H₀) by computing a Bayes factor via the Savage–Dickey density ratio (Dickey 1971). Specifically, a “null interval” for the introgression probability is defined as φ < ε within the H₁ parameter space is used to represent H₀. The Bayes factor, is approximated as B10, B_{10,\varepsilon}=\frac{\mathbb{P}(\varphi<\varepsilon)}{\mathbb{P}(\varphi<\varepsilon|X)}, where the numerator and denominator represent prior and posterior probabilities of φ < ε, respectively; as ε → 0, this converges to the true Bayes factor B_{10}. Following Ji et al. (Ji et al. 2023), we use ε = 0.01 and confirmed that ε = 0.001 yields similar results. As outlined above, a single significant D-statistic can be explained by multiple alternative introgression events—namely, gene flow between non-sister ingroup taxa (inflow or outflow) or ghost introgression from an unsampled lineage. Thus, for each significant signal, we test three possible introgression events: (1) introgression from P2 to P3 (outflow), (2) introgression from P3 to P2 (inflow), and (3) introgression from a ghost lineage to P1 (Fig. 1). Notably, the inflow and outflow are modeled jointly by a bidirectional introgression (Yang and Flouri, 2022), under the simplifying assumption that, if both directions occurred, they did so simultaneously. Only those with decisive support B₁₀ > 100 (analogous to a p-value threshold of 1% in frequentist hypothesis testing) are retained. 
Model construction proceeds iteratively. Starting with the highest-Dₚ triple, we incorporate the supported introgression events into the overall backbone tree. Horizontal reticulations (inflow and outflow) are placed on the branches of extant lineages, while ghost lineages are connected to the last common ancestor (LCA) of the involved taxa. Three exceptions warrant special handling: (1) When two triples share overlapping taxa, their introgression signals are jointly assessed and consolidated onto shared ancestral branches, following a parsimony-guided approach analogous to the f-branch method (Malinsky et al. 2021); (2) If multiple introgression events are inferred to fall on the same branch, their relative timing is determined by comparing the marginal likelihoods of the models with alternative orders (see step 4 below). (3) When the LCA is the ingroup root, all three positional scenarios of the ghost lineage relative to the outgroup (Fig. 2C–E) are also evaluated through marginal likelihood comparisons. Once a reticulation is incorporated for a given triple, any remaining triples whose signals are fully explained by existing events are excluded to avoid redundancy (Fig. 1); This process continues until all significant signals under the given species-tree topology are either incorporated or explained, yielding a finalized introgression model for that topology. The same steps are repeated for each starting tree, producing a set of introgression models corresponding to alternative species-tree topologies.

Step 4:
We compare the marginal likelihoods of introgression models constructed under each candidate species tree topology to identify the best-supported network. These marginal likelihoods are calculated in BPP via thermodynamic integration with Gaussian quadrature (Lartillot and Philippe 2006; Rannala and Yang 2017). The model with the highest marginal likelihood—representing the best-supported divergence and introgression scenario—is selected as the most probable under the MSci framework. Finally, we run the MCMC algorithm in BPP (Flouri et al. 2020) to obtain posterior parameter estimates for the best introgression model.


<img width="697" height="382" alt="image" src="https://github.com/user-attachments/assets/51e8c5e0-394d-44a7-8097-e214ec6f84a0" />


